public/controllers/management/components/management/common/actions-buttons.tsx

```diff
@@ -0,0 +1,186 @@
+/*
+ * Wazuh app - React component for registering agents.
+ * Copyright (C) 2015-2022 Wazuh, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Find more information about this on the LICENSE file.
+ */
+import React from 'react';
+// Eui components
+
+import { UploadFiles } from '../../upload-files';
+import { resourceDictionary, ResourcesHandler } from './resources-handler';
+import { WzButtonPermissions } from '../../../../../components/common/permissions/button';
+
+import { UI_ERROR_SEVERITIES } from '../../../../../react-services/error-orchestrator/types';
+import { UI_LOGGER_LEVELS } from '../../../../../../common/constants';
+import { getErrorOrchestrator } from '../../../../../react-services/common-services';
+
+/**
+ * Uploads the files
+ * @param {Array} files
+ * @param {String} path
+ */
+async function uploadFiles(files, resource, overwrite) {
+  try {
+    let errors = false;
+    let results: any[] = [];
+    const resourcesHandler = new ResourcesHandler(resource);
+    for (let idx in files) {
+      const { file, content } = files[idx];
+      try {
+        await resourcesHandler.updateFile(file, content, overwrite);
+        results.push({
+          index: idx,
+          uploaded: true,
+          file: file,
+          error: 0,
+        });
+      } catch (error) {
+        errors = true;
+        results.push({
+          index: idx,
+          uploaded: false,
+          file: file,
+          error: error,
+        });
+      }
+    }
+    if (errors) throw results;
+    return results;
+  } catch (error) {
+    throw error;
+  }
+}
+
+const getReadPermissionsFiles = (section) => {
+  const { permissionResource } = resourceDictionary[section];
+  return [
+    {
+      action: `${section}:read`,
+      resource: permissionResource('*'),
+    },
+  ];
+};
+
+const getUpdatePermissionsFiles = (section) => {
+  const { permissionResource } = resourceDictionary[section];
+  return [
+    {
+      action: `${section}:update`,
+      resource: permissionResource('*'),
+    },
+    {
+      action: `${section}:read`,
+      resource: permissionResource('*'),
+    },
+  ];
+};
+
+// Add new rule button
+export const AddNewFileButton = ({ section, updateAddingFile }) => (
+  <>{
+    section !== 'lists' &&
```

Comment:
**suggestion**: Maybe we should manage if the button is rendered in a higher level.